jupyterlab_search_replace/search_engine.py

```diff
@@ -122,7 +124,10 @@ async def search(
             query, max_count, case_sensitive, whole_word, include, exclude, use_regex
         )
         cwd = os.path.join(self._root_dir, url2path(path))
-        code, output = await self._execute(command, cwd=cwd)
+        if SearchEngine.search_task is not None:
```

Comment:
We probably don't want to cancel a done task neither.

```suggestion
        if SearchEngine.search_task is not None and not SearchEngine.done():
```